src/test/java/teammates/test/cases/browsertests/InstructorCourseDetailsPageUiTest.java

```diff
@@ -57,12 +64,86 @@ public void allTests() throws Exception {
         testTableSort();
         //No input validation required
         testLinks();
-        testRemindAction();
         testDeleteAction();
         testDeleteAllAction();
         testSanitization();
     }
 
+    @Test
+    public void testRemindAction() throws Exception {
+        instructorId = testData.instructors.get("CCDetailsUiT.instr").googleId;
+        courseId = testData.courses.get("CCDetailsUiT.CS2104").getId();
+        detailsPage = getCourseDetailsPage();
+        String courseName = testData.courses.get("CCDetailsUiT.CS2104").getName();
+        // student2 is yet to register, student1 is already registered
+        StudentAttributes student1 = testData.students.get("CCDetailsUiT.alice.tmms@CCDetailsUiT.CS2104");
+        EmailAccount student1EmailAccount = new EmailAccount(student1.email);
+        StudentAttributes student2 = testData.students.get("charlie.tmms@CCDetailsUiT.CS2104");
+        EmailAccount student2EmailAccount = new EmailAccount(student2.email);
+
+        boolean isEmailEnabled = !TestProperties.isDevServer();
+        if (isEmailEnabled) {
+            // get user authenticated so that we can access emails of them
+            student1EmailAccount.getUserAuthenticated();
+            student2EmailAccount.getUserAuthenticated();
+            // mark all emails as read to start from a clean state
+            student1EmailAccount.markAllUnreadEmailAsRead();
+            student2EmailAccount.markAllUnreadEmailAsRead();
+        }
+
+        ______TS("action: remind single student");
+
+        detailsPage.clickRemindStudentAndCancel(student2.name);
+        if (isEmailEnabled) {
+            assertStudentDoesNotReceiveReminder(courseName, courseId, student2EmailAccount);
+        }
+
+        detailsPage.clickRemindStudentAndConfirm(student2.name);
+        if (isEmailEnabled) {
+            assertStudentHasReceivedReminder(courseName, courseId, student2EmailAccount);
+            assertNull("getting registration key of the same course again should not return any result"
+                            + "as there should be only one reminder presented in the unread inbox",
+                    student2EmailAccount.getRegistrationKeyFromUnreadEmails(courseName, courseId));
+        }
+        detailsPage.waitForTextsForAllStatusMessagesToUserEquals(
+                Const.StatusMessages.COURSE_REMINDER_SENT_TO + student2.email);
+
+        // Hiding of the 'Send invite' link is already covered by content test.
+        //  (i.e., they contain cases of both hidden and visible 'Send invite' links.
+
+        ______TS("action: remind all");
+
+        detailsPage.clickRemindAllAndCancel();
```

Comment:
Missing check for whether student has received reminder after cancelling this action.